app/collections/show-list.cjsx

```diff
@@ -45,20 +47,32 @@ SubjectNode = React.createClass
 
     return hasPermission
 
+  handleSelect: (action) ->
+    @props.handleSelect @props.subject.id, action
+
   render: ->
     logClick = @context.geordi?.makeHandler? 'about-menu'
     <div className="collection-subject-viewer">
       <SubjectViewer defaultStyle={false} subject={@props.subject} user={@props.user} project={@state.project}>
-        {if @isOwnerOrCollaborator()
-          <button type="button" className="collection-subject-viewer-delete-button" onClick={@props.onDelete}>
-            <i className="fa fa-close" />
-          </button>}
-          <Link className="subject-link" to={"/projects/#{@state.project?.slug}/talk/subjects/#{@props.subject.id}"} onClick={logClick?.bind(this, 'view-favorite')}>
-            <span></span>
-          </Link>
+          {if !@props.selecting
+            <Link className="subject-link" to={"/projects/#{@state.project?.slug}/talk/subjects/#{@props.subject.id}"} onClick={logClick?.bind(this, 'view-favorite')}>
+              <span></span>
+            </Link>}
+          {if @isOwnerOrCollaborator() and !@props.selecting
+            <button type="button" className="collection-subject-viewer-delete-button" onClick={@props.onDelete}>
+              <i className="fa fa-close" />
+            </button>}
+          {if @props.selecting and !@props.selected
+            <button className="collection-subject-viewer-select-button" onClick={@handleSelect.bind @, 'add'}>
```

Comment:
Instead of two set of buttons, could a checkbox type input be considered with a single handler and the checked value be used to determine whether or not to add or remove? It could be wrapped in a label with the font awesome icons and a conditional to determine which to show based on the presence of the selected subject id in the state?